From 7a77ddb9a74513bb87525aac7d053f5b226d3314 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:59:39 -0400
Subject: [fs] btrfs: Add acl mount option

Message-id: <1407179109-13714-93-git-send-email-sandeen@redhat.com>
Patchwork-id: 87004
O-Subject: [RHEL7.1 PATCH 092/422] btrfs: Add acl mount option.
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Qu Wenruo <quwenruo@cn.fujitsu.com>

commit bd0330ad2174d1a5f762eee2ee58f0148f10d575
Author: Qu Wenruo <quwenruo@cn.fujitsu.com>
Date:   Mon Jan 6 09:58:30 2014 +0800

    btrfs: Add acl mount option.

    Add acl mount option to enable acl with remount option.

    Signed-off-by: Qu Wenruo <quwenruo@cn.fujitsu.com>
    Reviewed-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/filesystems/btrfs.txt b/Documentation/filesystems/btrfs.txt
index 4c8f6c3..0fc7ff2 100644
--- a/Documentation/filesystems/btrfs.txt
+++ b/Documentation/filesystems/btrfs.txt
@@ -141,8 +141,9 @@ Options with (*) are default options and will not show in the mount options.
  Specify that 1 metadata chunk should be allocated after every <value>
  data chunks.  Off by default.
 
+  acl(*)
   noacl
- Disable support for Posix Access Control Lists (ACLs).  See the
+ Enable/disable support for Posix Access Control Lists (ACLs).  See the
  acl(5) manual page for more information about ACLs.
 
   barrier(*)
diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index 98a6823..76eecd1 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -324,7 +324,7 @@ enum {
  Opt_check_integrity, Opt_check_integrity_including_extent_data,
  Opt_check_integrity_print_mask, Opt_fatal_errors, Opt_rescan_uuid_tree,
  Opt_commit_interval, Opt_barrier, Opt_nodefrag, Opt_nodiscard,
- Opt_noenospc_debug, Opt_noflushoncommit,
+ Opt_noenospc_debug, Opt_noflushoncommit, Opt_acl,
  Opt_err,
 };
 
@@ -347,6 +347,7 @@ static match_table_t tokens = {
  {Opt_ssd, "ssd"},
  {Opt_ssd_spread, "ssd_spread"},
  {Opt_nossd, "nossd"},
+ {Opt_acl, "acl"},
  {Opt_noacl, "noacl"},
  {Opt_notreelog, "notreelog"},
  {Opt_flushoncommit, "flushoncommit"},
@@ -552,6 +553,9 @@ int btrfs_parse_options(struct btrfs_root *root, char *options)
     goto out;
    }
    break;
+  case Opt_acl:
+   root->fs_info->sb->s_flags |= MS_POSIXACL;
+   break;
   case Opt_noacl:
    root->fs_info->sb->s_flags &= ~MS_POSIXACL;
    break;
-- 
1.7.1